### PR TITLE
Add Session Auth Deprecation Warning Message

### PR DESCRIPTION
**Pull request by andriantoniades**
This add a deprecation warning message when session authentication is used.

### CODE REVIEW

File: lib/fastly/client.rb [hunk 1]
```diff
@@ -57,6 +57,8 @@ def authed?
 
     # Some methods require full username and password rather than just auth token
     def fully_authed?
+      warn("DEPRECATION WARNING: Using username/password authentication is deprecated and will not be available, 
+      please migrate to API tokens as soon as possible.")
```
Comment (thommahoney) [inline lib/fastly/client.rb:61]: This method is called often and will end up littering customer logs with noisy output. I think we should move this warning into the `initialize` method and emit it when either `user` or `password` are provided.

File: test/fastly/client_test.rb [hunk 2]
```diff
@@ -29,6 +29,15 @@
       assert_equal "Invalid auth credentials. Check username/password.", e.message
     end
 
+    it 'surfaces a deprecation message when a username or password is provided' do
+      stub_request(:any, /api.fastly.com/).
+        to_return(body: JSON.generate(i: "dont care"), status: 200)
```
Comment (thommahoney) [inline test/fastly/client_test.rb:34]: I don't think these lines are necessary anymore. The tests should pass without the stubbing.

File: test/fastly/client_test.rb [hunk 3]
```diff
@@ -29,6 +29,15 @@
...
+
+      client = Fastly::Client.new(user: user, password: pass)
+      
+      assert_output('', /DEPRECATION WARNING:/) { $stderr.puts 'DEPRECATION WARNING:'}
```
Comment (thommahoney) [inline test/fastly/client_test.rb:38]: This test isn't really asserting anything unless you put the `Fastly::Client.new` invocation inside of the block.

```suggestion
      assert_output('', /DEPRECATION WARNING:/) { Fastly::Client.new(user: user, password: pass) }
```

File: lib/fastly/client.rb [hunk 4]
```diff
@@ -57,6 +61,10 @@ def authed?
 
     # Some methods require full username and password rather than just auth token
     def fully_authed?
+     has_both_username_and_password?
+    end
+
+    def has_both_username_and_password?
```
Comment (thommahoney) [inline lib/fastly/client.rb:67]: Since the two methods do exactly the same thing, I'd prefer to just have one of them. If you prefer the name `has_both_username_and_password?` better than `fully_authed?`, let's go ahead and rename the method—perhaps as a separate pull request. For now, would you mind reverting this change?

File: test/fastly/client_test.rb [hunk 5]
```diff
@@ -139,4 +148,4 @@
 
     end
   end
-end
+end
```
Comment (thommahoney) [inline test/fastly/client_test.rb:151]: This change was likely accidental. Let's revert this change too since you'll be changing this file.

File: test/fastly/client_test.rb (hunk 2 above)
Comment (andriantoniades) [reply test/fastly/client_test.rb:34]: The stub is required by Webmock
